src/libcore/iter/range.rs

```diff
@@ -267,14 +267,11 @@ impl<A: Step> ops::RangeFrom<A> {
     /// # Examples
     ///
     /// ```
-    /// # #![feature(step_by)]
+    /// #![feature(step_by)]
     ///
-    /// for i in (0u8..).step_by(2).take(10) {
-    ///     println!("{}", i);
-    /// }
+    /// let result: Vec<_> = (0..).step_by(2).take(5).collect();
+    /// assert_eq!(result, vec![0, 2, 4, 6, 8]);
```

Comment:
Yeah, sorry, I should have been clear. Removing the annotation isn't a no-op for the code, but it's unnecessary for explaining the behavior of `step_by`. A `u8` example may be important for explaining overflow behavior though...